OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -244,6 +246,8 @@ public virtual IEnumerable<ActorInfo> AllItems()
 
 		public virtual IEnumerable<ActorInfo> BuildableItems()
 		{
+			if (productionTraits.Any() && productionTraits.All(p => p.IsTraitDisabled))
```

Comment:
Minor perf nit (no blocker, I think): The `!Enabled` check below is certainly cheaper than LINQ, so I'd switch order of these two.